### PR TITLE
FF7Text: Process Remaining Characters

### PR DESCRIPTION
**Pull request by sithlord48**
- Handle All the Characters via FF7Text
 - Special Handling of 0xFEE2 (memory...) 
 - Special Handling of 0xFEDD {PAUSE%1}

@myst6re  Please check this over

### CODE REVIEW

File: src/data/FF7Text.cpp [hunk 1]
```diff
@@ -53,96 +60,152 @@ QString FF7Text::toPC(QByteArray text)
...
 
-    QString String;
+    QString str;
 
     for (quint16 i = 0 ; i < text.size() ; ++i) {
         quint8 index = quint8(text.at(i));
         if (index == 0xFF) {
             break;
+        } else if (index == 0XFE) {
+            if(i + 1 >= text.size()) {
```
Comment (myst6re) [inline src/data/FF7Text.cpp:70]:
```suggestion
            if (i + 1 >= text.size()) {
```

File: src/data/FF7Text.cpp [hunk 2]
```diff
@@ -53,96 +60,152 @@ QString FF7Text::toPC(QByteArray text)
...
+                continue;
+            } else {
```
Comment (myst6re) [inline src/data/FF7Text.cpp:72]: Else not needed, as the continue is enough to jump to the next iteration

File: src/data/FF7Text.cpp [hunk 3]
```diff
@@ -53,96 +60,152 @@ QString FF7Text::toPC(QByteArray text)
         text.truncate(txt);
```
Comment (myst6re) [inline src/data/FF7Text.cpp:60]: Using var assignment and comparison on the if clause is a bad practice. You can here set txt variable on the line 51

```cpp
int endIndex = text.indexOf('\xFF');
if (endIndex != -1) {
    text.truncate(endIndex);
}
```

File: src/data/FF7Text.cpp [hunk 4]
```diff
@@ -53,96 +60,152 @@ QString FF7Text::toPC(QByteArray text)
...
+                switch(quint8(text.at(i))) {
+                    case 0xDD:
+                        if (text.size() <= i + 1)
+                            break;
+                        ++i;
+                        str.append(QString("{PAUSE%1}").arg(quint8(text.at(i)), 3, 10, QChar('0')));
+                        continue;
+                    case 0xE2:
+                        if (i + 4 <= text.size() && quint8(text.at(i + 4)) == 0 && quint8(text.at(i + 2)) <= 4) {
+                            quint8 bank = quint8(text.at(i+2) > 4 );
```
Comment (myst6re) [inline src/data/FF7Text.cpp:83]:
```suggestion
                            quint8 bank = quint8(text.at(i + 2) > 4);
```

File: src/data/FF7Text.cpp [hunk 5]
```diff
@@ -53,96 +60,152 @@ QString FF7Text::toPC(QByteArray text)
...
+
+                            str.append(QString("{MEMORY:var[%2][%1];size=%3}")
+                                           .arg(quint8(text.at(i + 1)))
+                                           .arg(bank)
+                                           .arg(quint8(text.at(i + 3))));
+                            i += 4;
+                        }
+                        continue;
+                    default:
+                        str.append( (get()->isJapanese() || quint8(text.at(i)) >= 0xD2) ? character(text.at(i), 6) : QString(""));
```
Comment (myst6re) [inline src/data/FF7Text.cpp:101]:
```suggestion
                        str.append((get()->isJapanese() || quint8(text.at(i)) >= 0xD2) ? character(text.at(i), 6) : QString());
```

File: src/data/FF7Text.cpp (hunk 3 above)
Comment (sithlord48) [reply src/data/FF7Text.cpp:60]: Can Pause Wait or Memory have a 0xFF as a valid valid ?

Comment (sithlord48) [reply src/data/FF7Text.cpp]: No longer truncate at the first 0xFF values can be 255  as well

**Pull request merged**